src/controllers/services.controllers.js

```diff
@@ -1,26 +1,147 @@
-import { dataSource } from "../../db.js"
+import { dataSource } from "../../db.js";
+import { isValidType } from "../helpers/validateServiceType.js";
+import { uploadFile } from "../services/cloudinary.js";
 
-export const createNewCourse = async(req,res)=>{
-    console.log(req.body)
+const formationServiceRepository = dataSource.getRepository("FormationServices")
+
+
+export const createNewFormationService = async (req, res) => {
+
+  const type = req.body.type
+  if(!type || type && !isValidType(type)){
+    return res
+    .status(400)
+    .json({ isSuccess: false, message: "Please provide a valid type of course: 'course','seminar', 'workshop', 'diploma'" });
+  }
+  
+  
+  try {
     
-    const courseRepository = dataSource.getRepository("Course")
     let data = {
-        name:"Hola",
-        general_info:"Hola",
-        syllabus:"",
-        hours:20,
-        exhibitor_name:"Juan Carlos Montes",
-        exhibitor_photo:"www",
-        organized_by:"sadsad",
-        supported_by_name:"PYGLO",
-        supported_by_photo:"www",
-        cost:2500000,
-        discounts:"30% Egresados",
-        inscription_url:"www"
+      name: req.body.name,
+      general_info: req.body.general_info,
+      type:req.body.type,
+      image: req.body.image,
+      syllabus: req.body.syllabus,
+      hours: req.body.hours,
+      exhibitor_name: req.body.exhibitor_name,
+      organized_by: req.body.organized_by,
+      supported_by_name: req.body.supported_by_name,
+      cost: req.body.cost,
+      discounts: req.body.discounts,
+      inscription_url: req.body.inscription_url,
+    };
+
+    for (const imageField of Object.keys(req.files)) {
+      let file = req.files[imageField][0];
+      const res = await uploadFile(file.path);
+      data[imageField] = res.secure_url;
+    }
+
+    await formationServiceRepository.save(data)
+
+    return res.status(200).json({ isSuccess: true, message: "Created" });
+  } catch (error) {
+   
+    return res
+      .status(500)
+      .json({ isSuccess: false, message: "Ha ocurrido un error." });
+  }
+};
+
 
+export const getFormationServicesByPagination = async (req,res)=>{
+  const a = Number(req.params.taking)
+  try{
+    if(!req.params.skip || !req.params.taking || isNaN(Number(req.params.skip)) || isNaN(Number(req.params.taking))){
+      return res
+      .status(400)
+      .json({ isSuccess: false, message: "Please provide correct values for pagination. Example: /10/0. This will take 10 elements of each service, starting of id 1. You will get 40 total objects" });
```

Comment:
Suggestion: use DTO pattern to separate the validation logic from the query logic